lib/rbi/model.rb

```diff
@@ -605,9 +605,47 @@ def ==(other)
       visibility == other.visibility
     end
 
-    Public = T.let(Visibility.new(:public), Visibility)
-    Protected = T.let(Visibility.new(:protected), Visibility)
-    Private = T.let(Visibility.new(:private), Visibility)
+    sig { returns(T::Boolean) }
+    def public?
+      visibility == :public
+    end
+
+    sig { returns(T::Boolean) }
+    def protected?
+      visibility == :protected
+    end
+
+    sig { returns(T::Boolean) }
+    def private?
+      visibility == :private
+    end
+  end
+
+  class Public < Visibility
```

Comment:
I changed it here, but I think it's also true for `Attr`, `Param` and `Mixin`, I'll do the others in a separate PR.

Reply:
After playing with it a bit, it seems to not play well with class reopening: we can't reopen a children of a sealed class in another module.

I'm not going to use it since I use this pattern to split the concerns across the model, the printer, the rewriters etc.